Assets/Mirror/Runtime/NetworkConnection.cs

```diff
@@ -82,15 +82,17 @@ public void Disconnect()
             isReady = false;
             ClientScene.HandleClientDisconnect(this);
 
-            // paul:  we may be connecting or connected,  either way, we need to disconnect
-            // transport should not do anything if it is not connecting/connected
-            Transport.activeTransport.ClientDisconnect();
-
             // server? then disconnect that client
             if (Transport.activeTransport.ServerActive())
             {
                 Transport.activeTransport.ServerDisconnect(connectionId);
             }
+            else
```

Comment:
what about 'else if (Transport.activeTransport.ClientConnected())', just to be sure? since we check serveractive above too.